.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,37 @@
+# This action merges PRs that are tagged with `auto-merge`
+# once all checks have succeeded.
+
+name: "auto merge"
+
+on:
+  check_suite:
+    types: [completed]
+
+permissions:
+  contents: read
+
+jobs:
+  auto-merge:
+    permissions:
+      contents: write # for devmasx/merge-branch to merge branches
+      pull-requests: write # for peter-evans/create-or-update-comment to create or update comment
+    name: auto merge
+    if: github.repository_owner == 'NixOS' && github.event.pull_request.merged == false && contains(github.event.pull_request.labels.*.name, 'auto-merge') && github.event.check_suite.conclusion == 'success'
```

Comment:
I *think* that check_suite events contain the SHA of the commit for which they were ran. Maybe this is something we could use for merge-branch to additionally make sure that we're merging the right thing?